### ISSUE TITLE
Confirmed SNMPv3 encryption support AES128... created with --enable-blumenthal-aes not available in executables

### ISSUE THREAD

**Issue by FinzelStefan** [net-snmp/net-snmp#577]
NET-SNMP version: 5.9.3

Creating SNMPv3 encryption support AES128 AES192 AES192C AES256 AES256C with --enable-blumenthal-aes is confirmed in configuration summary but not available at all in executable command, e.g. snmpget

./configure --with-defaults -prefix=/opt/snmptest --enable-blumenthal-aes --disable-agent --disable-scripts --disable-snmptrapd-subagent --disable-debugging --disable-embedded-perl --disable-perl-cc-checks --disable-manuals --disable-static --enable-silent-libtool --with-cflags='-O3 -fomit-frame-pointer'

---------------------------------------------------------
            Net-SNMP configuration summary:
---------------------------------------------------------

  SNMP Versions Supported:    1 2c 3
  Building for:               linux
  Net-SNMP Version:           5.9.3
  Network transport support:  Callback Unix Alias TCP UDP TCPIPv6 UDPIPv6 IPv4Base SocketBase TCPBase UDPIPv4Base UDPBase IPBase IPv6Base
  SNMPv3 Security Modules:     usm
  Agent MIB code:            default_modules =>  snmpv3mibs mibII ucd_snmp notification notification-log-mib target agent_mibs agentx disman/event disman/schedule utilities host
  MYSQL Trap Logging:         unavailable
  Embedded Perl support:      disabled
  SNMP Perl modules:          building -- not embeddable
  SNMP Python modules:        disabled
  Crypto support from:        use_pkg_config_for_openssl
  Authentication support:     MD5 SHA1 SHA224 SHA256 SHA384 SHA512
  Encryption support:         DES AES AES128 AES192 AES192C AES256 AES256C
  Local DNSSEC validation:    disabled
...

> snmpget
NET-SNMP version: 5.9.3
...
  -a PROTOCOL		set authentication protocol (MD5|SHA|SHA-224|SHA-256|SHA-384|SHA-512)
 ...
 -x PROTOCOL		set privacy protocol (DES|AES)

**Comment by nielsb** [comment]
It works for me. Are you sure you run the proper binary, with the proper libraries?

**Comment by FinzelStefan** [comment]
Both binaries and libraries are there and working.
Playing around I found I have to add  --with-openssl=/usr in configure

./configure --with-defaults -prefix=/opt/tcl-sg --with-openssl=/usr --enable-blumenthal-aes --with-cflags='-O3 -fomit-frame-pointer'

It looks like it always requires --with-openssl=/usr on Ubuntu 22 and RHEL8.7: I already knew this for Rhaspian.

**Comment by nielsb** [comment]
The correct solution is to  install pkg-config

**Comment by FinzelStefan** [comment]
An up to date version of pkg-config is installed.

**Comment by nielsb** [comment]
I cannot replicate this with Debian 10 or 11 or Ubuntu 23.04. Can you provide the config.log where it fails?

**Comment by FinzelStefan** [comment]
I am sorry config.log is not accessible any more, it is on customer site. I assume everything is already deleted too, because I got it working at the end.
I've tested this at home with my developer system (Ubuntu 22.04) and some test systems (fresh installation of Ubuntu 22.04, RHEL 7.9, RHEL 8.7, RHEL 9.1, OpenSUSE LEAP 15.04).
It is fine with all(!!!) of them. None of them required --with-openssl=/usr, even RHEL 8.7 other than stated before. 
Could it be --with-defaults was fooled to find correct openssl path by additional openssl installations? There were several  required by other tools.
